packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -53,236 +60,283 @@ function findResponseByStatusCode(
     httpContent: IMediaTypeContent,
-  ): Omit<IHttpNegotiationResult, 'headers'> {
+  ): Either<Error, Omit<IHttpNegotiationResult, 'headers'>> {
     const { mediaType } = httpContent;
 
     if (exampleKey) {
       // the user provided exampleKey - highest priority
       const example = findExampleByKey(httpContent, exampleKey);
       if (example) {
         // example exists, return
-        return {
+        return right({
           code,
           mediaType,
           bodyExample: example,
-        };
+        });
       } else {
-        throw new Error(`Response for contentType: ${mediaType} and exampleKey: ${exampleKey} does not exist.`);
+        return left(new Error(`Response for contentType: ${mediaType} and exampleKey: ${exampleKey} does not exist.`));
       }
     } else if (dynamic === true) {
       if (httpContent.schema) {
-        return {
+        return right({
           code,
           mediaType,
           schema: httpContent.schema,
-        };
+        });
       } else {
-        throw new Error(`Tried to force a dynamic response for: ${mediaType} but schema is not defined.`);
+        return left(new Error(`Tried to force a dynamic response for: ${mediaType} but schema is not defined.`));
       }
     } else {
       // try to find a static example first
       const example = findBestExample(httpContent);
       if (example) {
         // if example exists, return
-        return {
+        return right({
           code,
           mediaType,
           bodyExample: example,
-        };
+        });
       } else if (httpContent.schema) {
-        return {
+        return right({
           code,
           mediaType,
           schema: httpContent.schema,
-        };
+        });
       } else {
-        return {
+        return right({
           code,
           mediaType,
-        };
+        });
       }
     }
   },
 
   negotiateDefaultMediaType(
     partialOptions: NegotiatePartialOptions,
     response: IHttpOperationResponse,
-  ): IHttpNegotiationResult {
+  ): Either<Error, IHttpNegotiationResult> {
     const { code, dynamic, exampleKey } = partialOptions;
     const httpContent =
       hasContents(response) &&
       (findDefaultContentType(response) || findBestHttpContentByMediaType(response, ['application/json']));
 
     if (httpContent) {
       // a httpContent for default mediaType exists
-      const contentNegotiationResult = helpers.negotiateByPartialOptionsAndHttpContent(
-        {
-          code,
-          dynamic,
-          exampleKey,
-        },
-        httpContent,
-      );
-      return {
-        headers: response.headers,
-        ...contentNegotiationResult,
-      };
+      return helpers
+        .negotiateByPartialOptionsAndHttpContent(
+          {
+            code,
+            dynamic,
+            exampleKey,
+          },
+          httpContent,
+        )
+        .map(contentNegotiationResult => ({
```

Comment:
hm.. why would mapping on `Either` be confusing?

Reply:
I understand @chris-miaskowski's confusion. I had the same when I started to look into this in November.

Thing is the `map `we use on daily basis on an arrays and objects is a specialisation of a more general map that is basically the introspection in an "effect".

In case of an `Array`, the effect is the "repetition" of values
In case of an `Object`, the effect is the presence of multiple properties
On an `Either` though, the effect is that "something went correctly". 

Array and Either are effectively wrapping/representing an effect.